### ISSUE TITLE
scikit-bio + simde v0.5.0: different numerical output

### ISSUE THREAD

**Issue by mr-c** [simd-everywhere/simde#474]
https://travis-ci.com/github/biocore/scikit-bio/jobs/351711278#L989

``` udiff
@@ -4,5 +4,5 @@
 number of groups                 2
 test statistic             1.03296
-p-value                       0.35
+p-value                       0.29
 number of permutations          99
 Name: PERMDISP results, dtype: object
```

**Comment by Artoria2e5** [comment]
Bit odd for plain SSE2 to go wrong. Does bumping up the submodule version help? Might be good to get something new from the CI.

FWIW, the PR is https://github.com/biocore/scikit-bio/pull/1709. Linking it so GitHuh would show a reference there too.

**Comment by nemequ** [comment]
I'm hoping that 63d87d9604e025816aed575afe6cf6a46d79635d fixes the problem, but haven't checked yet.  Other than that it's hard to think of what the problem could be.

If that doesn't fix the issue I'll probably have to examine the results of every simde call to look for something different than the native results, which really isn't something I'm looking forward to.

**Comment by mr-c** [comment]
@nemequ Thanks for that. I've rerun the tests with SIMDe 0.6.0 at https://travis-ci.com/github/biocore/scikit-bio/jobs/381064732

`[doctest] skbio.stats.distance._permdisp.permdisp` fails the exact same way as before: https://travis-ci.com/github/biocore/scikit-bio/jobs/381064732#L1135

**Comment by nemequ** [comment]
☹.  Okay, I'll try to take a look soon.

What is the p-value measuring here? I'm wondering if it's possible that this could be "failing" because SIMDe's results are actually *more* accurate.  We have several functions which can be significantly more accurate than the native versions; for example, rcp might be implemented as 1/x instead of an approximation. IIRC our portable rsqrt is also more accurate than the native version…

**Comment by mr-c** [comment]
I updated the scikit-bio PR to use a newer version of SIMDe and the issue went away. Thank you @nemequ and the other contributors!

The changelog of the differences since SIMDe 0.5.0 is at https://github.com/simd-everywhere/simde/releases/tag/v0.7.0-rc-1
Potential fixes could have been:

sse, sse2, sse4.1: fix ties-toward-even rounding 3208aeb @nemequ
sse4.1: better testing of _mm_round_ps b6a7310 @nemequ
sse: add simde_x_mm_round_ps with lax_rounding argument 24e5926 @nemequ
sse: fix NaN handling for _mm_max_ps, update test case 15aa0c4 @nemequ
sse2: add shuffle-based implementation of _mm_mul_epu32 e2da067 @nemequ
sse: add __builtin_nontemporal_store version of simde_mm_stream_ps 9a8001e @nemequ
